iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/memtable/WritableMemChunk.java

```diff
@@ -370,12 +373,37 @@ public void encode(IChunkWriter chunkWriter) {
         case TEXT:
         case BLOB:
         case STRING:
-          chunkWriterImpl.write(time, list.getBinary(sortedRowIndex));
+          Binary value = list.getBinary(sortedRowIndex);
+          chunkWriterImpl.write(time, value);
+          binarySizePerChunk += getBinarySize(value);
           break;
         default:
           LOGGER.error("WritableMemChunk does not support data type: {}", tsDataType);
           break;
       }
+      pointNumPerChunk++;
+      if (pointNumPerChunk > MAX_NUMBER_OF_POINTS_IN_CHUNK
+          || binarySizePerChunk > TARGET_CHUNK_SIZE) {
```

Comment:
TARGET_CHUNK_SIZE only consider binary types?
If I set MAX_NUMBER_OF_POINTS_IN_CHUNK to a very large value (like 1_000_000_000) and insert only number type, then TARGET_CHUNK_SIZE (like 1MB) will not take effect?

Reply:
The previous logic of flush triggered by TARGET_CHUNK_SIZE only considered the binary types.  Set MAX_NUMBER_OF_POINTS_IN_CHUNK to a very large value is a very weird behavior. I think in most of the common cases, TARGET_CHUNK_SIZE would take effect for binary types only. That's why the previous code only count the binary types I believe.

Reply:
One possible scenario is that I want to control the size of a chunk in byte so I set TARGET_CHUNK_SIZE to the value I want. 
However, I do not care and do not want to calculate the associated MAX_NUMBER_OF_POINTS_IN_CHUNK. Therefore, I just set MAX_NUMBER_OF_POINTS_IN_CHUNK to a very large value, which effectively disables the control so that chunks are only controlled by size.

Reply:
![image](https://github.com/user-attachments/assets/207dbe62-2af6-482b-b00d-cb0342756471)
The configuration says nothing about data types. I am fine if you insist, but at least we should give it a better name and explanation.